uwuno/src/main/java/com/learning/uwuno/room.java

```diff
@@ -84,8 +83,30 @@ public player getPlayer(String pid) {
         return playerList.stream().filter(t -> t.getPid().equals(pid)).findFirst().get();
     }
 
-    public void addPlayer(player newPlayer) {
+    // TODO: Need to change all the add players to use this instead? So names are checked
```

Comment:
We might not even be able to do this easily, I only have a fix for adding new players but for changing name will need to make a new function in room to handle change player name.

Reply:
- Should we have the hashmap in rooms? Otherwise it feels like we are redoing the same work.
- Should we make it so that the players can't change their name after they submit (i.e. is added)? It's like that for most online games we've played. So the only way to change name is if the person exits and rejoins.

Reply:
So have both the hashmap and normal list of players inside the room class? And if we do the double list thing inside the room, fixing changing name is pretty easy, just that having two lists is a little bit weird.

Reply:
What do you mean double list? And yeah, have both hashmap and normal list of players inside the room class so we just need to fetch the hashmap when addPlayer is called.

Reply:
hashmap + list = 2 lists of players. But yeah I can do that